paddle/pybind/pybind.cc

```diff
@@ -25,9 +26,143 @@ namespace pd = paddle::framework;
 
 USE_OP(add_two);
 
+struct PlaceDebugString : public boost::static_visitor<std::string> {
+  std::string operator()(const paddle::platform::GPUPlace& place) const {
+    return "GPU(" + std::to_string(place.device) + ")";
+  }
+
+  std::string operator()(const paddle::platform::CPUPlace& place) const {
+    return "CPU";
+  }
+};
+
+template <typename T>
```

Comment:
Also, `get_place` should be a public method for `tensor` because of two reason.

* The `CastToPyBufferImpl` class is in module `paddle::pybind`. If that class is a friend of `Tensor`, it could be
   1. Make cross dependency between two module.
   2. `CastToPyBufferImpl` is a `details` class, should be private in that module.
   3. `CastToPyBufferImpl` has many template arguments, and cannot be defined in one friend class.
* Tensor has an attribute of `place` actually. Even in mathematical thinking, tensor should not contain device information, but actually, in neural network frameworks, it actually holds a device information.